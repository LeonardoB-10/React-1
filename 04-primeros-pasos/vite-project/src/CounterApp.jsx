//Importamos la libreria para validar las propiedades
import PropTypes from "prop-types";
import { useState } from "react";

//Caundo se cambiar el estado se vuelve a renderizar el componente y se vuelve a ejecutar el codigo de la funcion del componente 
export const CounterApp = ({ value }) => {

    //[valor del estado, funcion para modificar el estado] = useState(valor inicial del estado)   
    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        //setCounter((c) => c + 1);
        setCounter(counter + 1);
    }

    const handleResta = () => {
        //setCounter((c) => c + 1);
        if(counter > 0){
          setCounter(counter - 1)
        }
    }  

    const valorInicial = () => {
        //setCounter((c) => c + 1);
        setCounter(value);
    } 

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleResta}>-1</button>
      <button aria-label="btn-resert" onClick={valorInicial}>Reset</button>
      {/* Se puede tomar el ID el arial-label */}
      
      {/* <button name="btn-resert" onClick={valorInicial}>Reset</button> */}
      {/* <button ="btidn-resert" onClick={valorInicial}>Reset</button> */}

    </>
  );
};

CounterApp.propTypes = {
  //Es obligatorio que se envie el parametro
  value: PropTypes.number.isRequired,
};
