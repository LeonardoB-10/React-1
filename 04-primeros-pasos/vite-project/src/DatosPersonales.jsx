//Todas las funciones iniciadas con mayuscula son componentes

//Se crea el componente y se le envia las propiedades que se van a usar en el componente
import PropTypes from "prop-types";

const  fechaNacimiento = () => {
    return <p>06/04/2000</p>
}

export const DaatosPersonales = ({ saludo,edad } ) =>{

  return (
    <>
      <h1>Hola</h1>
        <p>Nombre: Leonardo</p>
        <p>Apellido: Borja</p>
        {fechaNacimiento() }
        <p>{saludo}</p>
        <p>Con edad: {edad}</p>
    </>
  );
}

//Sirve para validar las propiedades que se le envian al componente
DaatosPersonales.propTypes = {
    //Es obligatorio que se envie el parametro
    saludo: PropTypes.string.isRequired,
    edad : PropTypes.number.isRequired
}

// Sirve para darle un valor por defecto a las propiedades
DaatosPersonales.defaultProps = {
    edad: 18,
    saludo : "Hola",
    name: "Leonardo"
}
//Exportamos el componente para poder usarlo en otros archivos

