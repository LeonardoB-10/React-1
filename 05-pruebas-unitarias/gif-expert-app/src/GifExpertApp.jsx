import { useState } from "react";
import { AddCategories } from "./componets/AddCategories";
import { GifGrid } from "./componets/GifGrid";

export const GifExpertApp = () => {

  const [categories,setcategories] = useState([]);

    const onAddCategory = (newCategory) => {
        // console.log(newCategory);
        if(categories.includes(newCategory)) return;//si la categoria ya existe no la agrega
        setcategories([newCategory,...categories]);//agrega la nueva categoria al principio del arreglo
        console.log(categories);
    }
     
    //se le pasa la funcion onAddCategory como prop al componente AddCategories
    return (
    <>
      <AddCategories onNewCategory = {event => onAddCategory(event)}/>
        {/* Entra la categoria */}
        {categories.map((category) => (
        <GifGrid
            key={category} 
            category={category}
            />
            ))}
    </>
  );
};
