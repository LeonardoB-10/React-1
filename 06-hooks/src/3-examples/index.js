
export * from './LoadingQuote';
export * from './MultipleCustonHooks';
export * from './Quote';