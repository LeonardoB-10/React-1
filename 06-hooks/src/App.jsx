// import { UserConuter} from './hooks/useCounter'

// function App() {

//   const {counter,incrementar,descrementar,reiniciar} = UserConuter();

//   return (
//     <div className="App"> 
//       <p>{counter}</p>
//          {/* (evento) => increment(evento) */}
//         <button onClick={()=>incrementar(2)}>+1</button>
//         <br />
//         <button onClick={()=>descrementar(2)}>-1</button> 
//         <br />
//         <button onClick={()=>reiniciar()}>Reiniciar</button>
//     </div>
//   )
// }

// export default App