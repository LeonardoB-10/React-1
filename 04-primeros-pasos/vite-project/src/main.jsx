import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'
import { DaatosPersonales } from './DatosPersonales'
import './style.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>

    <DaatosPersonales />
      {/* <CounterApp value={15} /> */}
  </React.StrictMode>
)
