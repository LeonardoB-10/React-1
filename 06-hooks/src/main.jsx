import React, { PureComponent } from 'react'
import ReactDOM from 'react-dom/client'
import { FormCustoHook } from './2-useEffect/FormCustoHosk'
import { SimpleForm } from './2-useEffect/SimpleForm'
import { MultibpleCustomHooks2 } from './3-examples/PruebaMia'
// import App from './App'


ReactDOM.createRoot(document.getElementById('root')).render(

    // <SimpleForm />
    // <FormCustoHook />
    <MultibpleCustomHooks2/>

)
