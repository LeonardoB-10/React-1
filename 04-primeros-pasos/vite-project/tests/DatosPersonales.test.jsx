import { DaatosPersonales } from "../src/DatosPersonales";
import { render } from "@testing-library/react";

describe("Pruebas en <FirstApp />", () => {
  test("debe de hacer march con el snapshot", () => {

    const saludo = "Hola";
    const edad = 18;
    
    //renderiza el componente y lo guarda en el contenedor de memoria
    const {container} = render(<DaatosPersonales saludo={saludo} edad={edad} />);
    //compara el snapshot con el componente actual y si no son iguales falla la prueba y crea un nuevo snapshot
    expect(container).toMatchSnapshot();
  });

  test("debe el titulo en un h1", () => {
    const saludo = "Hola";
    const edad = 18;
    //renderiza el componente y lo guarda en el contenedor de memoria
    const {container, getByText} = render(<DaatosPersonales saludo edad={edad} />);
    //verifica que el texto exista en el componente y no sea null
    expect(getByText(saludo)).toBeTruthy();

    const h1 = container.querySelector("h1");
    console.log(h1.innerHTML)
    
    // verifica que el texto sea igual al que se envio por parametro
    //expect(h1.innerHTML).toBe(saludo);

    expect(h1.innerHTML).toContain(saludo);
  });


});
