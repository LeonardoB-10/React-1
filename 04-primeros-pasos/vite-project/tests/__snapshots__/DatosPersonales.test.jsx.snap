// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer march con el snapshot 1`] = `
<div>
  <h1>
    Hola
  </h1>
  <p>
    Nombre: Leonardo
  </p>
  <p>
    Apellido: Borja
  </p>
  <p>
    06/04/2000
  </p>
  <p>
    Hola
  </p>
  <p>
    Con edad: 
    18
  </p>
</div>
`;
